rasa_core/channels/socketio.py

```diff
@@ -141,11 +137,24 @@ def disconnect(sid):
             logger.debug("User {} disconnected from socketio endpoint."
                          "".format(sid))
 
+        @sio.on('session_request', namespace=self.namespace)
+        def session_request(sid, data):
+            if data is None:
+                data = {}
+            if 'session_id' not in data or data['session_id'] is None:
+                data['session_id'] = str(uuid.uuid4())
+            sio.emit("session_confirm", data['session_id'])
+            logger.debug("User {} disconnected from socketio endpoint."
+                         "".format(sid))
+
         @sio.on(self.user_message_evt, namespace=self.namespace)
         def handle_message(sid, data):
-            output_channel = SocketIOOutput(sio, self.bot_message_evt)
-            message = UserMessage(data['message'], output_channel, sid,
-                                  input_channel=self.name())
-            on_new_message(message)
-
-        return socketio_webhook
+            output_channel = SocketIOOutput(sio, sid, self.bot_message_evt)
+            if "session_id" not in data or data["session_id"] is None:
+                logger.warning("A message without a valid sender_id was received")
```

Comment:
For this case, can we not use the sid as we did before?

Reply:
To avoid breaking changes I added a flag in the credentials and put a condition here.